### PR TITLE
Marian outreachform backup

### PR DESCRIPTION
**Pull request by marianbme**
changed db to outreacheventsdev

### CODE REVIEW

File: .DS_Store
Comment (phillipsk) [inline .DS_Store:1]: Remove this file

File: StreetCare/StreetCare/Views/Screens/OutreachFormView.swift [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    @State private var location = ""
+    @State private var date = Date()
+    @State private var time = Date()
+    @State private var eventDescription = ""
+    @State private var skillsDescription = ""  // Skills description field
+    @State private var showAlert = false
+    @State private var alertMessage = ""
+
+    private let db = Firestore.firestore()
+    private let titleLimit = 15
```
Comment (phillipsk) [inline StreetCare/StreetCare/Views/Screens/OutreachFormView.swift:22]: Expand this to 50 characters

**Review by phillipsk — CHANGES_REQUESTED** [review]
We need `Start Date` and `Start Time` as well as `End Date` and `End Time`

Location needs: Street, State, City, and Zipcode

Help Type Required section is missing
Maximum capacity is missing

Screenshot_20241114_193111


Screenshot_20241114_193331


Screenshot_20241114_193411

File: StreetCare/StreetCare/Views/Screens/OutreachFormView.swift [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    private let descriptionLimit = 200
+
+    var allFieldsFilled: Bool {
+        !title.isEmpty && !location.isEmpty
+    }
+
+    func saveToFirestore() {
```
Comment (phillipsk) [inline StreetCare/StreetCare/Views/Screens/OutreachFormView.swift:29]: add an `isLoading` state variable here:

```
    @State private var isLoading = false
    
    func saveToFirestore() {
        guard allFieldsFilled else {
            alertMessage = .emptyFields
            showAlert = true
            return
        }

        isLoading = true
        let formData: [String: Any] = [
            "title": title,
            "location": location,
            "date": date,
            "time": time,
            "eventDescription": eventDescription,
            "skillsDescription": skillsDescription
        ]

        db.collection("outreachEventsDev").addDocument(data: formData) { error in
            isLoading = false
            if let error = error {
                alertMessage = .error
                showAlert = true
            } else {
                alertMessage = .success
                showAlert = true
            }
        }
    }
```

and at the bottom of the VStack of the view add:

```
  .overlay(
      isLoading ? ProgressView("Saving...").padding().background(Color.white.opacity(0.8)).cornerRadius(10) : nil
  )

**Comment by marianbme** [comment]
Completed the changes except Enum Im getting complex view error because of that, creating new PR.

**Comment by marianbme** [comment]
New code updated

**Pull request closed without merging**